sdk/core/azure-core/src/http/transport_policy.cpp

```diff
@@ -17,27 +17,104 @@ using namespace Azure::Core::Http;
-  return ::AzureSdkGetCustomHttpTransport();
+    return ::AzureSdkGetCustomHttpTransport();
 #elif defined(BUILD_TRANSPORT_WINHTTP_ADAPTER)
-  return std::make_shared<Azure::Core::Http::WinHttpTransport>();
+    // Since C++11: If multiple threads attempt to initialize the same static local variable
+    // concurrently, the initialization occurs exactly once. We depend on this behavior to ensure
+    // that the singleton defaultTransport is correctly initialized.
+    static std::shared_ptr<HttpTransport> defaultTransport(std::make_shared<WinHttpTransport>());
+    if (AnyTransportOptionsSpecified(transportOptions))
+    {
+      WinHttpTransportOptions httpOptions;
+      if (!transportOptions.HttpProxy.empty())
+      {
+        std::string proxyString;
+        proxyString = "http=" + transportOptions.HttpProxy;
+        proxyString += ";";
+        proxyString += "https=" + transportOptions.HttpProxy;
+        httpOptions.ProxyInformation = proxyString;
+      }
+      httpOptions.ProxyUserName = transportOptions.ProxyUserName;
+      httpOptions.ProxyPassword = transportOptions.ProxyPassword;
+      return std::make_shared<Azure::Core::Http::WinHttpTransport>(httpOptions);
+    }
+    else
+    {
+      //      std::call_once(createTransportOnce, []() {
+      //      defaultTransport = std::make_shared<Azure::Core::Http::WinHttpTransport>();
+      //    });
+      return defaultTransport;
+    }
 #elif defined(BUILD_CURL_HTTP_TRANSPORT_ADAPTER)
-  return std::make_shared<Azure::Core::Http::CurlTransport>();
+    static std::shared_ptr<HttpTransport> defaultTransport(std::make_shared<CurlTransport>());
+    if (AnyTransportOptionsSpecified(transportOptions))
+    {
+      CurlTransportOptions curlOptions;
+      curlOptions.EnableCurlTracing = true;
+      if (!transportOptions.HttpProxy.empty())
+      {
+        curlOptions.Proxy = transportOptions.HttpProxy;
+      }
+      if (!transportOptions.ProxyUserName.empty())
+      {
+        curlOptions.ProxyUsername = transportOptions.ProxyUserName;
+      }
+      if (!transportOptions.ProxyPassword.empty())
+      {
+        curlOptions.ProxyPassword = transportOptions.ProxyPassword;
+      }
+      return std::make_shared<Azure::Core::Http::CurlTransport>(curlOptions);
+    }
+    return defaultTransport;
 #else
-  return std::shared_ptr<HttpTransport>();
+    return std::shared_ptr<HttpTransport>();
 #endif
+  }
+}}}}} // namespace Azure::Core::Http::Policies::_detail
+
+TransportPolicy::TransportPolicy(TransportOptions const& options) : m_options(options)
+{
+  // If there's no transport specified, then we need to create one.
+  // If there is one specified, it's an error to specify other options.
+  if (m_options.Transport)
+  {
+#if !defined(BUILD_TRANSPORT_CUSTOM_ADAPTER)
+    if (_detail::AnyTransportOptionsSpecified(options))
+    {
+      throw std::runtime_error(
+          "Invalid parameter: Proxies cannot be specified when a transport is specified.");
```

Comment:
This should be a precondition...  There's nothing people can do to recover from this on runtime..  this is an application error that should abort the app and must be fixed